pkg/controllers/vdb/podfacts.go

```diff
@@ -805,29 +744,45 @@ func (p *PodFacts) checkIsDBCreated(_ context.Context, vdb *vapi.VerticaDB, pf *
 	return nil
 }
 
-// makeNodeInfoFetcher will create a NodeInfoFetcher object based on the feature flags set
-// and the server version
+// makeNodeInfoFetcher will create a node-info Fetcher object based on the vclusterOps annotation
+// and the server version. If we are using vclusterops and server version is not older than v24.3.0,
+// the operator will call vclusterOps API to collect the node info. Otherwise, the operator will
+// execute vsql inside the pod to collect the node info.
 func (p *PodFacts) makeNodeInfoFetcher(vdb *vapi.VerticaDB, pf *PodFact) catalog.Fetcher {
-	return catalog.MakeVSQL(vdb, p.PRunner, pf.name, pf.execContainerName)
+	const vclusterAPISupportedMinVersion = "v24.3.0"
+	vdbVer, ok := vdb.GetVerticaVersionStr()
+	if ok {
```

Comment:
It might be good to log an entry if we cannot get the version. Just so we know why we reverted back to the old vsql behaviour.